helix-core/src/main/java/org/apache/helix/HelixPropertyFactory.java

```diff
@@ -88,8 +88,9 @@ private CloudConfig getCloudConfig(String zkAddress, String clusterName) {
     CloudConfig cloudConfig;
     RealmAwareZkClient dedicatedZkClient = null;
     try {
-      if (Boolean.getBoolean(SystemPropertyKeys.MULTI_ZK_ENABLED)) {
-        // If the multi ZK config is enabled, use multi-realm mode with DedicatedZkClient
+      if (Boolean.getBoolean(SystemPropertyKeys.MULTI_ZK_ENABLED) || zkAddress == null) {
```

Comment:
Seems we will need to follow this patten: 
`if (Boolean.getBoolean(SystemPropertyKeys.MULTI_ZK_ENABLED) || zkAddr == null)`
 in a couple of different places. Is there a better way to represent? The first one is self explanatory, but the second one does not look straightforward. The doc for explanation is also in another file.

Reply:
I have the comment explaining the behavior. Please suggest if you have another way to achieve this otherwise.

Reply:
Could you make it in a function? So for example, if later you have other conditions, you'll still need to add it in all related places?